runtime/vm/jvminit.c

```diff
@@ -2359,7 +2359,14 @@ IDATA VMInitStages(J9JavaVM *vm, IDATA stage, void* reserved) {
 			}
 
 			if (J2SE_VERSION(vm) >= J2SE_V11) {
-				vm->modularityPool = pool_new(OMR_MAX(sizeof(J9Package),sizeof(J9Module)),  0, 0, 0, J9_GET_CALLSITE(), J9MEM_CATEGORY_MODULES, POOL_FOR_PORT(vm->portLibrary));
+#if defined(J9VM_OPT_SNAPSHOTS)
+				if (IS_SNAPSHOT_RUN(vm)) {
+					vm->modularityPool = pool_new(OMR_MAX(sizeof(J9Package), sizeof(J9Module)),  0, 0, 0, J9_GET_CALLSITE(), J9MEM_CATEGORY_MODULES, POOL_FOR_PORT(VMSNAPSHOTIMPL_OMRPORT_FROM_JAVAVM(vm)));
+				} else if (!IS_RESTORE_RUN(vm))
```

Comment:
```suggestion
				if (IS_SNAPSHOT_RUN(vm)) {
					vm->modularityPool = pool_new(OMR_MAX(sizeof(J9Package), sizeof(J9Module)),  0, 0, 0, J9_GET_CALLSITE(), J9MEM_CATEGORY_MODULES, POOL_FOR_PORT(VMSNAPSHOTIMPL_OMRPORT_FROM_JAVAVM(vm)));
				} else if (IS_RESTORE_RUN(vm)) {
					/* vm->modularityPool is already initialized as part of the restore */
				} else
```

Make the 3 cases {snapshot, restore, normal jvm} explicit.